pkg/portproxy/server.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright © 2024 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package portproxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/rancher-sandbox/rancher-desktop-agent/pkg/types"
+	"github.com/rancher-sandbox/rancher-desktop-networking/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const bridgeIPAddr = "192.168.1.2"
+
+type PortProxy struct {
+	address  string
+	listener net.Listener
+	quit     chan struct{}
+	// map of port number as a key to associated listener
+	activeListeners map[int]net.Listener
+	mutex           sync.Mutex
+	wg              sync.WaitGroup
+}
+
+func NewPortProxy(address string) (*PortProxy, error) {
```

Comment:
Rather than passing in an address, passing in a `net.Listener` for the socket might be easier (especially for testing).